interfaces/apparmor/backend.go

```diff
@@ -256,6 +264,27 @@ func addContent(securityTag string, snapInfo *snap.Info, opts interfaces.Confine
 	}
 }
 
+// procSelfMountInfo is exposed as a variable for testing anyEncryptedDirectory.
+var procSelfMountInfo = mount.ProcSelfMountInfo
+
+// anyEncryptedDirectory returns true ecryptfs is mounted anywhere.
+// TODO: Make this look for the place where the snap is mounted from, once that
+// information becomes available.
```

Comment:
"once that information becomes available"? At the time of 'snap try' invocation we know the user and can therefore determine if we should add the snippets. We just need to store that off somewhere so that future refreshes can check on whether or not to add these snippets. This is most correct but also makes the feature robust in the face of snap refresh happening when no users are logged in. (Not to mention, ecryptfs might be used on the system for something other than encrypted home and this would unconditionally add the weakened policy to the snap when it doesn't need to).